Add gitignore travis.yml and setup.py + fix linux_test

svalentin opened the pull request:
- gitignore helps us ignore compiled files and tox test envs.
- .travis.yml will be used by travis to run Continuous Integration
  tests on new pull requests.
- setup.py will be used to later integrate the lib to pip. Other tools
  will then be able to depend on this lib and have dependencies
  automatically pulled in.
- while adding the tests, I noticed that linux_test was failing. The
  order of the dev listing was probably undetermined. Fixed by sorting
  the list. Also moved the fs initialization to setup to make sure there
  are no leaks between tests.

gdasher commented on setup.py:
```diff
@@ -0,0 +1,58 @@
+                     'interacting with a U2F device over USB.',
+    url='https://github.com/google/pyu2f/',
+    author='Google Inc.',
+    author_email='gnubby-team@google.com',
```
Lets make this pyu2f-team@google.com, which I just created with relevant membership

svalentin replied:
Done!

svalentin commented:
PTAL!
Fixed the travis build. It was travis.yml not travis.yaml.
Also fixed linux_test to always pass. It was flaky apparently due to the listing of devices being of undetermined order.
Changed email by amending the commit so the old email should not be in any history.

svalentin commented:
ping!
